ibis/expr/signature.py

```diff
@@ -187,25 +189,24 @@ def __eq__(self, other) -> bool:
         return type(self) == type(other) and self.args == other.args
 
     def __getstate__(self) -> Dict[str, Any]:
-        return {
-            key: getattr(self, key)
-            for key in self.__signature__.parameters.keys()
-        }
+        return {key: getattr(self, key) for key in self.argnames}
+
+    @property
+    def args(self):
+        if not (result := self._args):
```

Comment:
isn't this 3.10 only syntax? or is there compat providing this

Reply:
3.8: https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions